autorest/azure/auth/auth.go

```diff
@@ -41,59 +41,100 @@ import (
 // 3. Username password
 // 4. MSI
 func NewAuthorizerFromEnvironment() (autorest.Authorizer, error) {
-	tenantID := os.Getenv("AZURE_TENANT_ID")
-	clientID := os.Getenv("AZURE_CLIENT_ID")
-	clientSecret := os.Getenv("AZURE_CLIENT_SECRET")
-	certificatePath := os.Getenv("AZURE_CERTIFICATE_PATH")
-	certificatePassword := os.Getenv("AZURE_CERTIFICATE_PASSWORD")
-	username := os.Getenv("AZURE_USERNAME")
-	password := os.Getenv("AZURE_PASSWORD")
-	envName := os.Getenv("AZURE_ENVIRONMENT")
-	resource := os.Getenv("AZURE_AD_RESOURCE")
-
-	var env azure.Environment
-	if envName == "" {
-		env = azure.PublicCloud
-	} else {
-		var err error
-		env, err = azure.EnvironmentFromName(envName)
-		if err != nil {
-			return nil, err
-		}
+	settings, err := GetAuthSettings()
+	if err != nil {
+		return nil, err
+	}
+
+	if settings.Resource == "" {
+		settings.Resource = settings.Environment.ResourceManagerEndpoint
+	}
+
+	return settings.GetAuth()
+}
+
+// NewAuthorizerFromEnvironmentWithResource creates an Authorizer configured from environment variables in the order:
+// 1. Client credentials
+// 2. Client certificate
+// 3. Username password
+// 4. MSI
+func NewAuthorizerFromEnvironmentWithResource(resource string) (autorest.Authorizer, error) {
+	settings, err := GetAuthSettings()
+	if err != nil {
+		return nil, err
 	}
+	settings.Resource = resource
+	return settings.GetAuth()
+}
+
+// Settings reporesnt settings for authentication
+type Settings struct {
```

Comment:
im not sure it provides any value to expose this extra type since you added the `NewAuthorizerFromEnvironmentWithResource` .